### ISSUE TITLE
How to use in my own dataset?

### ISSUE THREAD

**Issue by Jacobew** [wutong16/DistributionBalancedLoss#10]
Hi, how can I use DistributionBalancedLoss on my own dataset?

**Comment by harshaldesai01** [comment]
> Hi, how can I use DistributionBalancedLoss on my own dataset?

@Jacobew , were you able to use it in your dataset?

**Comment by wutong16** [comment]
Hi, sorry for the late reply!

To apply it in your own datasets, you may:
1. Merge your datasets to this project: define the `dataset Class` following the coco/voc/lvis style like in [here](https://github.com/wutong16/DistributionBalancedLoss/tree/master/mllt/datasets), and then import the `Class` here in [this file](https://github.com/wutong16/DistributionBalancedLoss/blob/a3ecaa9021a920fcce9fdafbd7d83b51bf526af8/mllt/datasets/__init__.py). Then you could set the name of the new dataset in the config file (`dataset_type=name_of_the_custom_dataset`) to load it.
2. Another way is to adopt BOTH the **re-sampling module** and **loss function** to your own project.

**Comment by harshaldesai01** [comment]
Thank you @wutong16 for that. 

Also, I don't understand why reduction is expicitly set to 'none' [here](https://github.com/wutong16/DistributionBalancedLoss/blob/a3ecaa9021a920fcce9fdafbd7d83b51bf526af8/mllt/models/losses/resample_loss.py#L122) even after doing override [here](https://github.com/wutong16/DistributionBalancedLoss/blob/a3ecaa9021a920fcce9fdafbd7d83b51bf526af8/mllt/models/losses/resample_loss.py#L109)

**Comment by wutong16** [comment]
Hi @Jacobew,

Notice the `if-else` condition in that part of the function: the parameter `reduction` is used [here](https://github.com/wutong16/DistributionBalancedLoss/blob/a3ecaa9021a920fcce9fdafbd7d83b51bf526af8/mllt/models/losses/resample_loss.py#L127) in the `else` branch. While in the `if self.focal` branch, the `reduction` is set as 'none' to produce raw logits in the form of a vector to further compute the focal loss.
